File: C#/09-Laço Do While/Fatorial.cs
using System;

    class Fatorial
    {
        static void Main(string[] args)
        {
            //Agora vamos implementar um algoritmo para calculo de fatorial usando Do While
            long c, n, f;
            string resp;
    //Este primeiro controla o numero de operações
            do
            {
                Console.WriteLine("Digite um número :");
                n = long.Parse(Console.ReadLine());
                c = n;
                f = 1;
                do
                {
                    Console.Write(c +" X ");
                    f = f * c;
                    c = c - 1;
                } while(c >= 1);

                Console.WriteLine("\nO fatorial de " +n +" é " +f);
                Console.WriteLine("Deseja fazer outra operação? s/n");
                resp = Console.ReadLine();
            } while(resp != "n");
        }
    }

File: Java/09-Laço Do While/Fatorial.java
import java.util.Scanner;

class Fatorial {
    public static void main(String args[]) {
        // Podemos usar o do while em uma algoritmo para calculo de fatorial
        long c, n, f;
        String resp;
        Scanner entrada = new Scanner(System.in);

        // Vamos usar dois do while
        do {
            System.out.println("Digite um numero :");
            n = entrada.nextLong();
            c = n;
            f = 1;

            do {
                System.out.print(c + " X ");
                f = f * c;
                c -= 1;
            } while (c >= 1);

            System.out.println("\nO fatorial de " + n + " e " + f);
            System.out.println("Deseja fazer outra operacao? [s/n]");
            resp = entrada.next();
        } while (resp.equals("s"));

        entrada.close();
    }
}